### PR TITLE
Add a quickstart guide

### LINKED ISSUE #4: Create an initial quickstart guide

### PR DESCRIPTION
**Pull request by Sharpz7**
Closes #4

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+FastTrack can be run using the following command:
+
+```bash
+docker run --rm -p 5000:5000 -ti fasttrack
+```
+
+Verify that you can see the UI by navigating to http://localhost:5000/.
+
+![FastTrack UI](https://files.mcaq.me/57b05.jpg)
+
```
Comment (jgiannuzzi) [inline docs/quickstart.md:38]: we probably just want to show how to _run_ fasttrack in the quickstart, which can be done with the DockerHub image (`gresearch/fasttrack`)

Comment (Sharpz7) [reply docs/quickstart.md:38]: Yeah, good point. I'll remove the build steps and make that a seperate doc :))

File: docs/dev/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[tool.poetry]
+name = "fast-track-test"
```
Comment (kannon92) [inline docs/dev/pyproject.toml:2]: I was pretty disappointed with poetry. Could we just use setup tools like we do in python client?

File: docs/developer.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# Install json parser
+sudo apt-get install jq
+
+# Get the build tags and version from the settings.json file
+tags="$(jq -r '."go.buildTags"' .vscode/settings.json)"
+version=$(git describe --tags | sed 's/^v//')
+
+docker build --build-arg tags="$tags" --build-arg version="$version" -t fasttrack .
+
+## Build FastTrack
```
Comment (kannon92) [inline docs/developer.md:26]: Duplicate section?

File: docs/dev/pyproject.toml (hunk 2 above)
Comment (Sharpz7) [reply docs/dev/pyproject.toml:2]: Honestly I am the opposite. I would rather work out the headaches of using poetry with the Python Client than what we currently do. Maybe we can discuss this further?

File: docs/developer.md (hunk 3 above)
Comment (Sharpz7) [reply docs/developer.md:26]: Thanks, I will fix this now :)

File: docs/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# FastTrack Quickstart
+
+## Install Dependencies
+
+FastTrack requires the following dependencies to be installed on your system:
+
+- [Docker](https://docs.docker.com/get-docker/)
+- [Docker Buildx](https://docs.docker.com/buildx/working-with-buildx/)
```
Comment (jgiannuzzi) [inline docs/quickstart.md:8]: not required for running

File: docs/dev/data/0.csv [hunk 5]
```diff
@@ -0,0 +1,2910 @@
+26.0,4.0,5.0,8.0,-1.0,-13.0,-109.0,-66.0,-9.0,2.0,4.0,13.0,-18.0,-30.0,-119.0,-45.0,17.0,6.0,-1.0,-9.0,27.0,20.0,91.0,71.0,-26.0,-1.0,0.0,1.0,13.0,20.0,-62.0,-19.0,29.0,2.0,2.0,0.0,-23.0,-1.0,-80.0,4.0,-7.0,-6.0,-12.0,-27.0,-11.0,-16.0,-67.0,-8.0,-27.0,1.0,1.0,13.0,-8.0,-11.0,21.0,-28.0,61.0,4.0,8.0,5.0,4.0,-7.0,-59.0,16.0,0
```
Comment (kannon92) [inline docs/dev/data/0.csv:1]: Do you think we should pull these down as temp files?  They bloat your repo and all you really need is some examples..

Maybe just `wget` from kaggle as part of a script and ignore these files?

File: docs/dev/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+version = "0.1.0"
+description = "A Random Forrest Classifier for EMG Data - Fast Track Test"
```
Comment (kannon92) [inline docs/dev/pyproject.toml:4]:
```suggestion
description = "A Random Forest Classifier for EMG Data - Fast Track Test"
```

File: docs/dev/minimal.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from random import randint, random
+
+import mlflow
+from mlflow import log_metric, log_param, sklearn
```
Comment (jgiannuzzi) [inline docs/dev/minimal.py:4]:
```suggestion
from mlflow import log_metric, log_param
```

File: docs/dev/minimal.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+mlflow.set_tracking_uri("http://localhost:5000/mlflow")
```
Comment (jgiannuzzi) [inline docs/dev/minimal.py:6]:
```suggestion
mlflow.set_tracking_uri("http://localhost:5000")
```

File: docs/dev/random_forrest.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+import mlflow
+
+ALL_FILES = ["data/0.csv", "data/1.csv", "data/2.csv", "data/3.csv"]
+
+mlflow.set_tracking_uri("http://localhost:5000/mlflow")
```
Comment (jgiannuzzi) [inline docs/dev/random_forrest.py:9]:
```suggestion
mlflow.set_tracking_uri("http://localhost:5000")
```

File: docs/developer.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+FastTrack can be built using the following command:
+
+```bash
+# Install json parser
+sudo apt-get install jq
+
+# Get the build tags and version from the settings.json file
+tags="$(jq -r '."go.buildTags"' .vscode/settings.json)"
+version=$(git describe --tags | sed 's/^v//')
```
Comment (jgiannuzzi) [inline docs/developer.md:22]:
```suggestion
version=$(git describe --tags --dirty | sed 's/^v//')
```

File: docs/developer.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# Developer Guide
+
+## Install Dependencies
+
+FastTrack requires the following dependencies to be installed on your system:
+
+- [Docker](https://docs.docker.com/get-docker/)
+- [Docker Buildx](https://docs.docker.com/buildx/working-with-buildx/)
+- [Python 3](https://www.python.org/downloads/)
+- [MLFlow](https://mlflow.org/docs/latest/index.html)
```
Comment (jgiannuzzi) [inline docs/developer.md:10]: MLFlow is not required for development

File: docs/developer.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
+# Developer Guide
```
Comment (jgiannuzzi) [inline docs/developer.md:1]: We should probably have 2 sections:
- devcontainer
- "classic" (currently covered in your doc)

File: docs/quickstart.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+FastTrack can be run using the following command:
+
+```bash
+docker run --rm -p 5000:5000 -ti gresearch/fasttrack
+```
+
+Verify that you can see the UI by navigating to http://localhost:5000/.
+
+![FastTrack UI](https://files.mcaq.me/57b05.jpg)
```
Comment (jgiannuzzi) [inline docs/quickstart.md:22]: Could you include this file in your PR?
(Same remark for all the other screenshots)

File: docs/quickstart.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+python3 ./docs/dev/minimal.py
+```
+
+After running this script, you should see the following output from http://localhost:5000/aim/:
+
+![FastTrack UI](https://files.mcaq.me/43x5j.jpg)
+
+From here you can check out the metrics and run information to see more details about the run.
+
+## Testing a Random Forrest Model
```
Comment (jgiannuzzi) [inline docs/quickstart.md:37]:
```suggestion
## Testing a Random Forest Model
```

File: docs/quickstart.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+**Note that since artifacts are not yet supported, most of the autolog features will not work.**
+
+```bash
+# Install mflow and poetry
+cd docs/dev
+poetry install
+# MLFlow will not be installed by poetry, so we need to install it manually
```
Comment (jgiannuzzi) [inline docs/quickstart.md:45]: Why is it not installed by poetry?

File: docs/quickstart.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+
+Verify that you can see the UI by navigating to http://localhost:5000/.
+
+![FastTrack UI](https://files.mcaq.me/57b05.jpg)
+
+
+## Run a quick test script
+
+```bash
+python3 ./docs/dev/minimal.py
```
Comment (jgiannuzzi) [inline docs/quickstart.md:28]: This won't work without first installing MLFlow

File: docs/quickstart.md (hunk 13 above)
Comment (Sharpz7) [reply docs/quickstart.md:22]: Ah. Yeah good point.

File: docs/quickstart.md (hunk 15 above)
Comment (Sharpz7) [reply docs/quickstart.md:45]: It broke on my system, and I can't find any refence to people doing it successfully

File: docs/dev/data/0.csv (hunk 5 above)
Comment (Sharpz7) [reply docs/dev/data/0.csv:1]: I don't think this is a trivial process... Kaggle make it very difficult.

Comment (kannon92) [reply docs/dev/data/0.csv:1]: Okay

**Review by kannon92 — COMMENTED** [review]
LGTM but I’ll leave it to @jgiannuzzi for final look.

**Review by jgiannuzzi — COMMENTED** [review]
Could you please remove the csv files from the repo and rename `docs/dev` to `docs/example`. Integrating the quickstart guide directly into the README would be good too (removing the current `Building` section and replacing the `Running` one).

File: docs/quickstart.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+## Install Dependencies
+
+FastTrack requires the following dependencies to be installed on your system:
+
+- [Docker](https://docs.docker.com/get-docker/)
+- [Python 3](https://www.python.org/downloads/)
+
+```
+
```
Comment (jgiannuzzi) [inline docs/quickstart.md:11]:
```suggestion

```

**Comment by Sharpz7** [comment]
Sorry for the delay, I think that is the changes implemented :))

File: README.md [hunk 18]
```diff
@@ -5,17 +5,20 @@ Rewrite of the MLFlow tracking server with a focus on scalability
...
+
+```bash
+docker run --rm -p 5000:5000 -ti gresearch/fasttrack
 ```
-docker run --rm -ti fasttrack
-```
+
+Verify that you can see the UI by navigating to http://localhost:5000/.
+
+![FastTrack UI](images/main_ui.jpg)
```
Comment (jgiannuzzi) [inline README.md:21]:
```suggestion
![FastTrack UI](docs/images/main_ui.jpg)
```

**Review by jgiannuzzi — APPROVED** [review]
Thanks a lot for your contribution @Sharpz7!

**Pull request merged**